src/viam/sdk/components/encoder/server.hpp

```diff
@@ -0,0 +1,52 @@
+/// @file components/encoder/server.hpp
+///
+/// @brief Implements a gRPC server for the `Encoder` component.
+#pragma once
+
+#include <viam/api/common/v1/common.pb.h>
+#include <viam/api/component/encoder/v1/encoder.grpc.pb.h>
+
+#include <viam/sdk/resource/resource_server_base.hpp>
+#include <viam/sdk/subtype/subtype.hpp>
+
+namespace viam {
+namespace sdk {
+
+/// @class EncoderServer
+/// @brief gRPC server implementation of a `Encoder` component.
+/// @ingroup Encoder
+class EncoderServer : public ResourceServerBase,
+                      public viam::component::encoder::v1::EncoderService::Service {
+   public:
+    ::grpc::Status GetPosition(
+        ::grpc::ServerContext* context,
+        const ::viam::component::encoder::v1::GetPositionRequest* request,
+        ::viam::component::encoder::v1::GetPositionResponse* response) override;
+
+    ::grpc::Status ResetPosition(
+        ::grpc::ServerContext* context,
+        const ::viam::component::encoder::v1::ResetPositionRequest* request,
+        ::viam::component::encoder::v1::ResetPositionResponse* response) override;
+
+    ::grpc::Status GetProperties(
+        ::grpc::ServerContext* context,
+        const ::viam::component::encoder::v1::GetPropertiesRequest* request,
+        ::viam::component::encoder::v1::GetPropertiesResponse* response) override;
+
+    ::grpc::Status DoCommand(grpc::ServerContext* context,
+                             const viam::common::v1::DoCommandRequest* request,
+                             viam::common::v1::DoCommandResponse* response) override;
+
+    void register_server(std::shared_ptr<Server> server) override;
+
+    std::shared_ptr<SubtypeService> resource_manager();
```

Comment:
Just to confirm: has `cryvex` been updated to not include `ResourceManager`/`resource_manager()` in server code in the future?

Reply:
I did this because of this comment https://github.com/viamrobotics/viam-cpp-sdk/pull/91#issuecomment-1528082450 . This was a temporary solution to make it so that updating this class post #91 is much easier.